fideslog/api/routes/events.py

```diff
@@ -0,0 +1,135 @@
+from logging import getLogger
+
+from fastapi import APIRouter, Depends, HTTPException, Request, status
+from sqlalchemy.exc import DBAPIError
+from sqlalchemy.orm import Session
+
+from fideslog.api.config import config
+from fideslog.api.database.data_access import create_event
+from fideslog.api.database.database import get_db
+from fideslog.api.schemas.analytics_event import AnalyticsEvent
+
+log = getLogger(__name__)
+event_router = APIRouter(tags=["Events"], prefix="/events")
+
+
+@event_router.post(
+    "",
+    response_description="The created event",
+    response_model=AnalyticsEvent,
+    responses={
+        status.HTTP_429_TOO_MANY_REQUESTS: {
+            "content": {
+                "application/json": {
+                    "example": {
+                        "error": f"Rate limit exceeded: {config.server.request_rate_limit}"
+                    },
+                    "schema": {
+                        "type": "object",
+                        "properties": {"error": {"type": "string"}},
+                    },
+                }
+            },
+            "description": "Rate limit exceeded",
+            "headers": {
+                "Retry-After": {
+                    "description": "The datetime after which to retry the request.",
+                    "schema": {"type": "http-date"},
+                },
+                "X-RateLimit-Limit": {
+                    "description": "The number of allowed requests in the current period.",
+                    "schema": {"type": "integer"},
+                },
+                "X-RateLimit-Remaining": {
+                    "description": "The number of remaining requests in the current period.",
+                    "schema": {"type": "integer"},
+                },
+                "X-RateLimit-Reset": {
+                    "description": "The number of seconds left in the current period.",
+                    "schema": {"type": "integer"},
+                },
+            },
+        }
```

Comment:
This is rewritten in 3 places. It would be nice to DRY it up by defining a constant in an `api/errors.py` file, like:
```py
# api/errors.py

from logging import getLogger
from typing import Dict, Union

from fastapi import HTTPException, status

from .config import config

log = getLogger(__name__)


class TooManyRequestsError(HTTPException):
    """
    To be raised when a request exceeds the configured rate limit.
    """

    MESSAGE = "Rate limit exceeded"

    def __init__(self, error: Exception) -> None:
        log.error("%s: %s", self.MESSAGE, error, exc_info=error)
        super().__init__(
            status.HTTP_429_TOO_MANY_REQUESTS,
            detail={"message": self.MESSAGE, "limit": config.server.request_rate_limit},
        )

    @classmethod
    def doc(cls) -> Dict[str, Union[Dict, str]]:
        """
        Returns the documentation for a 429 Too Many Requests response,
        in the OpenAPI spec format.
        """

        return {
            "content": {
                "application/json": {
                    "example": {
                        "error": f"Rate limit exceeded: {config.server.request_rate_limit}"
                    },
                    "schema": {
                        "type": "object",
                        "properties": {"error": {"type": "string"}},
                    },
                }
            },
            "description": "Rate limit exceeded",
            "headers": {
                "Retry-After": {
                    "description": "The datetime after which to retry the request.",
                    "schema": {"type": "http-date"},
                },
                "X-RateLimit-Limit": {
                    "description": "The number of allowed requests in the current period.",
                    "schema": {"type": "integer"},
                },
                "X-RateLimit-Remaining": {
                    "description": "The number of remaining requests in the current period.",
                    "schema": {"type": "integer"},
                },
                "X-RateLimit-Reset": {
                    "description": "The number of seconds left in the current period.",
                    "schema": {"type": "integer"},
                },
            },
        }

```
Then, here + everywhere else this is done, you can instead just do:
```py
from ..errors import TooManyRequestsError
```
```suggestion
        status.HTTP_429_TOO_MANY_REQUESTS: TooManyRequestsError.doc()
```